src/IO/ReadWriteBufferFromHTTP.cpp

```diff
@@ -216,6 +216,9 @@ void ReadWriteBufferFromHTTPBase<UpdatableSessionPtr>::getHeadResponse(Poco::Net
             if (i == settings.http_max_tries - 1 || !isRetriableError(response.getStatus()))
                 throw;
 
+            if (e.code() == ErrorCodes::TOO_MANY_REDIRECTS)
+                throw;
+
```

Comment:
```suggestion
            if (i == settings.http_max_tries - 1 || e.code() == ErrorCodes::TOO_MANY_REDIRECTS !isRetriableError(response.getStatus()))
                throw;
```

Reply:
It's okay to have a few `if (cond) throw;` statements instead of `if (very || long || condition || that  || (may_be && !easy_to_read)) throw;`. Also, there's `||` missing in the suggested change